### PR TITLE
[stable/3.0] Backport various pre-check commits to 3.0

### PR DESCRIPTION
**Pull request by jsuchome**
Mostly it's backport of https://github.com/crowbar/crowbar-core/pull/729 + parts of https://github.com/crowbar/crowbar-core/pull/689/commits/905dbae00e267e21fdeda314eae76b11536e4e5d

### CODE REVIEW

**Comment by jsuchome** [comment]
[(5) S49P9 #upgrade Synchronize API responses with the Angular app](https://trello.com/c/3TALQwDA/121-5-s49p9-upgrade-synchronize-api-responses-with-the-angular-app)

File: crowbar_framework/app/models/api/upgrade.rb [hunk 1]
```diff
@@ -202,11 +234,15 @@ def compute_resources_check_errors
...
         {
           compute_resources: {
-            data: [], # TODO: implement cluster health check errors
+            data: compute_resources_status[:errors],
             help: I18n.t("api.upgrade.prechecks.compute_resources_check.help.default")
           }
         }
       end
+
+      def ha_presence_check
```
Comment (MaximilianMeister) [inline crowbar_framework/app/models/api/upgrade.rb]: I think you have to either drop this method, as it is not used, or reference it from `ha_presence_status`

File: crowbar_framework/spec/models/api/crowbar_spec.rb [hunk 2]
```diff
@@ -195,7 +195,48 @@
...
         and_return(exit_code: 0, stdout: "HEALTH_WARN", stderr: "")
       )
-      expect(subject.class.ceph_healthy?).to be false
+      expect(subject.class.ceph_status).to_not be_empty
+    end
+  end
+
+  context "with HA deployed" do
+    it "succeeds to confirm that HA is deployed" do
+
```
Comment (MaximilianMeister) [inline crowbar_framework/spec/models/api/crowbar_spec.rb]: can you drop this empty line? same below

File: crowbar_framework/app/models/api/crowbar.rb [hunk 3]
```diff
@@ -127,27 +127,46 @@ def clusters_healthy?
...
+              "Found only one compute node of #{virt} type; non-disruptive upgrade is not possible"
+            )
+          end
         end
-        true
+      end
+
+      # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+      def ha_presence_check
+        ret = {}
```
Comment (MaximilianMeister) [inline crowbar_framework/app/models/api/crowbar.rb]: would you mind using `tap` here, as in `ceph_status` and `compute_resource_status`?

Comment (jsuchome) [inline crowbar_framework/app/models/api/crowbar.rb]: Actually... I'll make it the same as in master branch (https://github.com/crowbar/crowbar-ha/blob/master/crowbar_framework/app/models/api/pacemaker.rb#L21) which is probably different commit that I used here, so I wont call it cherry pick any more

**Comment by skazi0** [comment]
lgtm :)

**Pull request merged**